addons/mass_mailing/models/mailing.py

```diff
@@ -345,11 +406,23 @@ def create(self, vals_list):
                 values['name'] = "%s %s" % (values['subject'], now)
             if values.get('body_html'):
                 values['body_html'] = self._convert_inline_images_to_urls(values['body_html'])
-        return super().create(vals_list)
+        mailings = super().create(vals_list)
+        campaign_vals = []
+        for mailing in mailings:
+            if mailing.is_ab_testing and not mailing.campaign_id:
+                campaign_vals.append(mailing._get_default_ab_testing_campaign_values())
+        self.env['utm.campaign'].create(campaign_vals)
+        return mailings
 
     def write(self, values):
         if values.get('body_html'):
             values['body_html'] = self._convert_inline_images_to_urls(values['body_html'])
+        if values.get('ab_testing_schedule_datetime'):
+            cron = self.env.ref('mass_mailing.ir_cron_mass_mailing_ab_testing').sudo()
+            at = fields.Datetime.from_string(values['ab_testing_schedule_datetime'])
+            cron._trigger(at=at)
+        if values.get('campaign_id') is False and self.is_ab_testing and 'is_ab_testing' not in values:
```

Comment:
If I just hit "ab testing" at write, isn't it creating a campaign like at create ? Is it intended ?